src/solvers/smt2/smt2_solver.cpp

```diff
@@ -262,6 +262,42 @@ void smt2_solvert::command(const std::string &c)
       }
       std::cout << ')' << '\n';
     }
+    else if(c == "get-model")
+    {
+      // print a model for all identifiers
+
+      if(status != SAT)
+        throw error("model is not available");
+
+      bool first = true;
+
+      std::cout << '(';
+      for(const auto &id : id_map)
+      {
+        const symbol_tablet symbol_table;
+        const namespacet ns(symbol_table);
```

Comment:
Hopefully the compiler optimises this, but creating those in each loop iteration seems wasteful. Maybe just do it outside the loop?

Reply:
ok, will hoist